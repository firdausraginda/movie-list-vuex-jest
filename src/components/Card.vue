<template>
  <div class="card col-4">
    <div class="card-body">
      <img
        class="card-img-top"
        :src="dataCard.imgFilm"
        alt="Card image cap"
      >
      <h5 class="card-title">{{cutTextJudul(dataCard.judulFilm)}}</h5>
      <p>{{cutTextDeskripsi(dataCard.deskripsiFilm)}}</p>
      <a
        class="btn btn-primary"
        @click="moveToDetail()"
      >Detail</a>
      <a
        v-if="dataCard.choosen == false"
        class="btn"
        @click="chooseDataFil()"
      >Pilih</a>
      <a
        v-if="this.$route.name == 'personal' && dataCard.choosen == true"
        class="btn"
        @click="deleteDataFil()"
      >Hapus</a>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      idFilmx: ""
    };
  },
  props: {
    dataCard: {
      type: Object,
      default: null
    }
  },
  methods: {
    cutTextJudul(text) {
      return `${text.substr(0, 30)}`;
    },
    cutTextDeskripsi(text) {
      return `${text.substr(0, 100)}..`;
    },
    moveToDetail() {
      this.$router.push({ path: `/detailFilm/${this.dataCard.idFilm}` });
    },
    chooseDataFil() {
      this.$store.dispatch("addDataFilmChoosen", this.dataCard);
    },
    deleteDataFil() {
      this.$store.dispatch("deleteDataFilmChoosen", this.dataCard);
    }
  }
};
</script>

<style lang="scss" scoped>
.btn-primary {
  color: white !important;
}
.card {
  border: none;
  text-align: left;
  img {
    border-radius: 5px;
    height: 200px;
    object-fit: cover;
    margin-bottom: 20px;
  }

  .card-title {
    font-weight: bold;
  }
}
</style>