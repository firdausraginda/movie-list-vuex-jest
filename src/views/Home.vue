<template>
  <div class="home">
    <h2>{{title}}</h2>
    <div class="container">
      <div class="row">
        <Card
          v-for="film in getDaftarFilm"
          :key="film.idFilm"
          :dataCard="film"
        />
      </div>
    </div>
    <!-- <img alt="Vue logo" src="../assets/logo.png"> -->
    <!-- <HelloWorld msg="Welcome to Your Vue.js App"/> -->
  </div>
</template>

<script>
// @ is an alias to /src
import Card from "../components/Card.vue";

export default {
  data() {
    return {
      title: "Daftar Film"
    };
  },
  components: {
    Card
  },
  computed: {
    getDaftarFilm() {
      return this.$store.state.daftarFilm;
    }
  }
};
</script>