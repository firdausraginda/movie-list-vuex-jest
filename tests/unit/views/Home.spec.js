import { shallowMount } from '@vue/test-utils'
import Home from '../../../src/views/Home.vue'
import Card from '../../../src/components/Card.vue'

describe('Home.spec.js', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(Home, {
            computed: {
                getDaftarFilm() {
                    return [{ idFilm: 'iu1kh291', choosen: false, imgFilm: 'https://upload.wikimedia.org/wikipedia/en/thumb/5/53/Black_mirror_bandersnatch_poster.jpg/220px-Black_mirror_bandersnatch_poster.jpg', judulFilm: 'Black Mirror: Bandersnatch', deskripsiFilm: 'Black Mirror: Bandersnatch adalah film interaktif tahun 2018 dalam seri antologi fiksi ilmiah Black Mirror. Film ini ditulis oleh pencipta serinya, Charlie Brooker, dan disutradarai oleh David Slade. Netflix merilisnya sebagai film terpisah pada tanggal 28 Desember 2018.' }]
                }
            }
        })
    })

    test('default', () => {
        expect(wrapper.is(Home)).toBe(true)
    })

    test('default2', () => {
        expect(wrapper.find(Card).vm.dataCard.idFilm).toEqual('iu1kh291')
    })

    test('has "title" string type of data with "Daftar Film" value', () => {
        expect(typeof wrapper.vm.title).toBe('string')
        expect(wrapper.vm.title).toEqual('Daftar Film')
    })

    test('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

})