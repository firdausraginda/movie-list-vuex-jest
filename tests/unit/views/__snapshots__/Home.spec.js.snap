// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.spec.js has the expected html structure 1`] = `
<div
  class="home"
>
  <h2>
    Daftar Film
  </h2>
   
  <div
    class="container"
  >
    <div
      class="row"
    >
      <card-stub
        datacard="[object Object]"
      />
    </div>
  </div>
</div>
`;
